storage/engine/rocksdb/db.cc

```diff
@@ -645,17 +649,13 @@ bool MergeValues(cockroach::storage::engine::MVCCMetadata *left,
       rocksdb::Warn(logger, "inconsistent value types for merge (left = bytes, right = ?)");
       return false;
     }
-    // Check for replay; if left merge_timestamp is equal to right,
-    // then ignore the merge request. Replays can happen on merges as
-    // there is no higher-level replay protection and Raft may
-    // duplicate commands. The replay protection here is imperfect.
-    // Raft may duplicate and replay commands out of order, in which
-    // case the same command may be re-merged.
-    //
-    // NOTE: there are test cases in storage/engine/merge_test and in
-    //   ts/db_test which will send MVCCMetadata with empty
-    //   merge_timestamps and should always be merged; thus the check
-    //   for non-empty merge_timestamps.
+
+    // Replay Advisory: Because merge commands pass through raft, it is possible
+    // for merging values to be "replayed". Currently, the only actual use of
+    // the merge system is for time series data, which is safe against replay;
+    // however, this property is not general for all potential mergeable types.
+    // If a future need arises to merge another type of data, replay protection
+    // will likely need to be a consideration.
     if (left->has_merge_timestamp() && right.has_merge_timestamp()) {
       if (left->merge_timestamp().wall_time() == right.merge_timestamp().wall_time() &&
           left->merge_timestamp().logical() == right.merge_timestamp().logical()) {
```

Comment:
No, you were right the first time, I wanted to remove this.  I had a weird merge at the very end, didn't delete enough.